apps/passport-server/src/services/devconnect/organizerSync.ts

```diff
@@ -0,0 +1,760 @@
+          eventSettingErrors.join("\n")
+      );
+    }
+
+    for (const item of items) {
+      // Ignore items which are not in the event's "activeItemIDs" set
+      if (activeItemIdSet.has(item.id.toString())) {
+        const itemErrors = this.validateEventItem(item);
+        if (itemErrors.length > 0) {
+          errors.push(
+            `Product "${item.name.en}" (${item.id}) in event "${eventData.eventInfo.name.en}" is invalid:\n` +
+              itemErrors.join("\n")
+          );
+        }
+      }
+    }
+
+    return errors;
+  }
+
+  /**
+   * Validate fetched data, or throw an exception
+   */
+  private validate(fetchedData: EventDataFromPretix[]): void {
+    const errors = [];
+
+    for (const { data, event } of fetchedData) {
+      errors.push(...this.checkEventData(data, event));
+    }
+
+    if (errors.length > 0) {
+      for (const error of errors) {
+        logger(
+          `[DEVCONNECT PRETIX]: Encountered error when validating fetched data for ${this.organizer.id}: ${error}`
+        );
+      }
+      throw new Error(errors.join("\n"), { cause: errors });
+    }
+  }
+
+  /**
+   * Fetch all of the API responses from Pretix necessary to sync an event,
+   * so that we can inspect them before beginning a sync.
+   */
+  private async fetchEventData(
+    organizer: DevconnectPretixOrganizerConfig,
+    event: DevconnectPretixEventConfig
+  ): Promise<EventData> {
+    return traced(NAME, "fetchEventData", async () => {
+      const { orgURL, token } = organizer;
+      const { eventID } = event;
+
+      const settings = await this.pretixAPI.fetchEventSettings(
+        orgURL,
+        token,
+        eventID
+      );
+
+      const items = await this.pretixAPI.fetchItems(orgURL, token, eventID);
+
+      const eventInfo = await this.pretixAPI.fetchEvent(orgURL, token, eventID);
+
+      const tickets = await this.pretixAPI.fetchOrders(orgURL, token, eventID);
+
+      return { settings, items, eventInfo, tickets };
+    });
+  }
+
+  /**
+   * Sync a single event.
+   * This coordinates the syncing of event info, items, and tickets to the DB.
+   * No actual fetching from Pretix happens here, as the data was already
+   * fetched when checking for validity.
+   */
+  private async saveEvent(
+    organizer: DevconnectPretixOrganizerConfig,
+    event: DevconnectPretixEventConfig,
+    eventData: EventData
+  ): Promise<void> {
+    return traced(NAME, "saveEvent", async (span) => {
+      try {
+        const { eventInfo, items, tickets } = eventData;
+
+        span?.setAttribute("org_url", organizer.orgURL);
+        span?.setAttribute("ticket_count", tickets.length);
+        span?.setAttribute("event_slug", eventInfo.slug);
+        span?.setAttribute("event_name", eventInfo.name.en);
+
+        if (!(await this.syncEventInfos(organizer, event, eventInfo))) {
+          logger(
+            `[DEVCONNECT PRETIX] Aborting sync due to error in updating event info`
+          );
+          return;
```

Comment:
maybe this and the 2 matching sections below can throw rather than log + return, so that the errors bubble up to error reporting?

Reply:
I've made this change. It only throws one level up, though - there's a try/catch in the same function.